### PR TITLE
Added controller for request and ooo requests API

### PR DESCRIPTION
**Pull request by sahsisunny**
**Description:**
In this pull requests added controller function for OOO requests with requests API.

**TEST:** Integration test is added in #https://github.com/Real-Dev-Squad/website-backend/pull/1843 PR

### CODE REVIEW

File: controllers/oooRequests.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { createOooRequest } from "../models/oooRequests";
+import { OooStatusRequestResponse, OooStatusRequestRequest } from "../types/oooRequest";
+import { addLog } from "../models/logs";
+import { CREATE, ERRORS, ERROR_WHILE_CREATING_OOO_REQUEST, OOO_LOG_TYPE, OOO_STATUS_REQUEST_CREATED_SUCCESSFULLY } from "../constants/oooRequest";
+
+export const createOooRequestController = async (req: OooStatusRequestRequest, res: OooStatusRequestResponse) => {
+    const oooRequestBody = req.body;
+    const userId = req.userData.id;
```
Comment (iamitprakash) [inline controllers/oooRequests.ts:8]: what if req is null or userData is null

File: controllers/oooRequests.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    try {
+        const oooRequestDoc = await createOooRequest({userId, ...oooRequestBody});
```
Comment (iamitprakash) [inline controllers/oooRequests.ts:11]: oooRequestDoc please use better name

File: controllers/oooRequests.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        if ('error' in oooRequestDoc) {
+            const oooRequestLog = {
+                type: OOO_LOG_TYPE.OOO_REQUEST_BLOCKED,
+                meta: {
+                    action: ERRORS,
+                    createdBy: userId,
+                    createdAt: Date.now(),
```
Comment (iamitprakash) [inline controllers/oooRequests.ts:18]: don't we need to convert date into specific timezone

File: controllers/requests.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { OooStatusRequestResponse, OooStatusRequestRequest } from "../types/oooRequest";
+import { REQUEST_TYPE } from "../constants/request";
+import { createOooRequestController } from "./oooRequests";
+
+export const createRequestController = async (
+  req: OooStatusRequestRequest,
```
Comment (iamitprakash) [inline controllers/requests.ts:6]: please change this

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: controllers/oooRequests.ts (hunk 1 above)
Comment (sahsisunny) [reply controllers/oooRequests.ts]: Done , added null check condition

File: controllers/oooRequests.ts (hunk 2 above)
Comment (sahsisunny) [reply controllers/oooRequests.ts]: Done

File: controllers/requests.ts (hunk 4 above)
Comment (sahsisunny) [reply controllers/requests.ts]: Done

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**